util/util.go

```diff
@@ -68,6 +68,25 @@ func RemoveSortedDuplicates(sortedIn []int64) []int64 {
 	return result
 }
 
+func CountAndRemoveSortedDuplicates(sortedIn []int64) ([]int64, map[int64]int64) {
```

Comment:
I think `RemoveSortedDuplicates` can be redefined to take advantage of this new function and eliminate some duplicate code.

Something like this:
```
func RemoveSortedDuplicates(sortedIn []int64) []int64 {
 	result, counts := CountAndRemoveSortedDuplicates(sortedIn)
 	return result
 }
```

Reply:
While the code is cleaner, doing this would result in poor performance in the beaconing analyzer.

`data.ts = util.RemoveSortedDuplicates(data.ts)`
This function is called to dedup any timestamps that come through. These duplications are rare. Since duplicates are rare, the counts map would end up containing almost as many entries as the `data.ts` array. Maps necessarily require more storage than arrays to hold the same elements. Creating and throwing away that much data in a hot section of the code seems rather costly.

CountAndRemoveSortedDuplicates is used to dedup the timestamp delta times and the connection data sizes. Thankfully, the timestamp delta times tend to clump together, creating fewer records in the count map. Unfortunately, the data sizes are not as clustered and consume more ram in the deduplication process. 

Thoughts?

Reply:
That's a good point.  Efficiency is definitely a concern, and with the size of the datasets we're dealing with I'd say it's more important than clean code.

Quick back of the napkin calculation:
A `map[int64]int64` will be at least 16 bytes (8 for each int64).  If n is 100,000 then it's 1.6 MB.  Network bandwidth scales on orders of magnitudes (e.g. 100 Mbit, 1 Gb, 10 Gb, etc) so I'm going to assume connection count could do the same. It's not out of the realm of possibility to deal with 2 or even 3 magnitudes larger in the future which would be 160 Mb and then 1.6 GB.  That does seem quite wasteful on memory.

I agree it's safer to err on the side of performance.  Someday if we have the ability to solidly measure performance benchmarks it will be safer to make changes like this knowing the actual impact.